src/components/new-safe/steps/Step0/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Button, Divider, Grid, Paper, Typography } from '@mui/material'
+import useWallet from '@/hooks/wallets/useWallet'
+import ChainSwitcher from '@/components/common/ChainSwitcher'
+import NetworkSelector from '@/components/common/NetworkSelector'
+import useIsWrongChain from '@/hooks/useIsWrongChain'
+import { useCurrentChain } from '@/hooks/useChains'
+import { isPairingSupported } from '@/services/pairing/utils'
+
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import WalletDetails from '@/components/common/ConnectWallet/WalletDetails'
+import PairingDetails from '@/components/common/PairingDetails'
+import useCreateSafe from '@/components/new-safe/CreateSafe/useCreateSafe'
+import type { ConnectedWallet } from '@/services/onboard'
+
+export const ConnectWalletContent = ({ onSubmit }: { onSubmit: StepRenderProps<NewSafeFormData>['onSubmit'] }) => {
+  const isWrongChain = useIsWrongChain()
+  const wallet = useWallet()
+  const chain = useCurrentChain()
+  const isSupported = isPairingSupported(chain?.disabledWallets)
+
+  const onConnect = (wallet?: ConnectedWallet) => {
```

Comment:
I feel like the wallet should not be allowed to be undefined or if we have to allow this we should check for this before we call  `onSubmit`
```
if (wallet) {
  onSubmit({ owners: [{ address: wallet.address || '', name: wallet.ens || '' }] })
}
```

Reply:
Added a check for the wallet before calling onSubmit